### PR TITLE
Workspaceapi

### PR DESCRIPTION
**Pull request by rajivnathan**
## Description
Workspace is the Schema for the workspaces API but it is only for use by the Proxy. There will be no actual Workspace CRs in the host/member clusters. The CRD will be installed in member clusters for API discovery purposes only. The schema will be used by the proxy's workspace lister API.

## Checks
1. Did you run `make generate` target? **yes**

2. Did `make generate` change anything in other projects (host-operator, member-operator)? **yes**

3. In case of **new** CRD, did you the following? **yes**
    - make/generate.mk in this repository
    - `resources/setup/roles/host.yaml` in the sandbox-sre repository
    - `PROJECT` file: https://github.com/codeready-toolchain/host-operator/blob/master/PROJECT
    - `CSV` file: https://github.com/codeready-toolchain/host-operator/blob/master/config/manifests/bases/host-operator.clusterserviceversion.yaml

4. In case other projects are changed, please provides PR links.
    - member-operator: https://github.com/codeready-toolchain/member-operator/pull/406

### CODE REVIEW

**Review by alexeykazakov — APPROVED** [review]

File: api/v1alpha1/workspace_types.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// WorkspaceStatus defines the observed state of a Workspace
+// +k8s:openapi-gen=true
+type WorkspaceStatus struct {
+	// The list of namespaces belonging to the Workspace.
+	// +listType=atomic
```
Comment (MatousJobanek) [inline api/v1alpha1/workspace_types.go:11]:
```suggestion
	// +optional
	// +listType=atomic
```

File: api/v1alpha1/workspace_types.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	Role string `json:"role,omitempty"`
+}
+
+// WorkspaceNamespace the information about a namespace within a Workspace
+// +k8s:openapi-gen=true
+type WorkspaceNamespace struct {
+
+	// Name the name of the namespace.
+	Name string `json:"name"`
+}
```
Comment (MatousJobanek) [inline api/v1alpha1/workspace_types.go:32]: do we want to add also the `type` field  to match the one that will be available at Space CRs? https://redhat-internal.slack.com/archives/CHK0J6HT6/p1674810629791589

Comment (rajivnathan) [reply api/v1alpha1/workspace_types.go:33]: Is it something we want to expose via the space lister? @alexeykazakov

Comment (alexeykazakov) [reply api/v1alpha1/workspace_types.go:33]: Yes, we need `type` here too. Good catch @MatousJobanek

Comment (rajivnathan) [reply api/v1alpha1/workspace_types.go]: Added in https://github.com/codeready-toolchain/api/pull/337/commits/1dca22722d9d82d16853d0fc6304471bb698710b

File: api/v1alpha1/spacenamespaces.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package v1alpha1
+
+// SpaceNamespace is a common type to define the information about a namespace within a Space
+// Used in NSTemplateSet status and Workspace status
+// +k8s:openapi-gen=true
```
Comment (alexeykazakov) [inline api/v1alpha1/spacenamespaces.go:5]: We don't need this, do we?
```suggestion
```

Comment (rajivnathan) [reply api/v1alpha1/spacenamespaces.go]: Yeah, don't think we need it. Removed: https://github.com/codeready-toolchain/api/pull/337/commits/81bf04f3c5d3bfefe41812ff5b28766f1dd41a01

**Review by alexeykazakov — APPROVED** [review]

File: api/v1alpha1/spacenamespaces.go [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package v1alpha1
+
+// SpaceNamespace is a common type to define the information about a namespace within a Space
+// Used in NSTemplateSet status and Workspace status
```
Comment (mfrancisc) [inline api/v1alpha1/spacenamespaces.go:4]:
```suggestion
// Used in NSTemplateSet, Space and Workspace status
```

**Review by mfrancisc — APPROVED** [review]
Looks good! I left only one small suggestion.

**Review by MatousJobanek — APPROVED** [review]

**Pull request merged**